src/embeddings/embeddings_api.hpp

```diff
@@ -0,0 +1,34 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "rapidjson/document.h"
+
+enum class EncodingFormat {
+    FLOAT,
+    BASE64
+};
+
+struct EmbeddingsRequest {
+    std::variant<std::vector<std::string>, std::vector<std::vector<int>>> input;
```

Comment:
it could be also std::string or vector<int>

Reply:
in this case a singular string or vector would be represented by either one vector with size 1 simplify declaration of the variant since std::variant<std::vector<std::string>, std::vector<std::vector<int>>, std::string, std::vector<int>> might be unreadable.